### PR TITLE
chore: upgrade swappers to structurally support multi-hop trades

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

Upgrades swappers to support multi-hop trades structurally, without actually implementing multi-hop trades in the application. Also removes redundant `sendMax` and `maximumCryptoHuman` from swappers.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

NA

## Risk

This is a fairly high risk change.
- Risk of trades generally broken
- Risk of broken max trades

## Testing

This change should introduce zero change from production.

- check that quotes are being created correctly including fees and receive amounts
- check that trades execute correctly for all swappers
- check that max trades execute correctly for all swappers

### Engineering

See above

### Operations

This change should introduce zero change from production.

### CODE REVIEW

File: src/lib/swapper/api.ts [hunk 1]
```diff
@@ -149,16 +148,21 @@ interface TradeBase<C extends ChainId, MissingNetworkFee extends boolean = false
...
+> & {
   allowanceContract: string
+}
+
+export type TradeQuote<C extends ChainId, UnknownNetworkFee extends boolean = false> = {
   minimumCryptoHuman: string
-  maximumCryptoHuman: string
   recommendedSlippage?: string
   id?: string
+  steps: TradeQuoteStep<C, UnknownNetworkFee>[]
```
Comment (woodenfurniture) [inline src/lib/swapper/api.ts:162]: This is the main change

File: src/lib/swapper/api.ts [hunk 2]
```diff
@@ -149,16 +148,21 @@ interface TradeBase<C extends ChainId, MissingNetworkFee extends boolean = false
...
+export type TradeQuoteStep<C extends ChainId, UnknownNetworkFee extends boolean> = TradeBase<
+  C,
+  UnknownNetworkFee
+> & {
   allowanceContract: string
+}
+
+export type TradeQuote<C extends ChainId, UnknownNetworkFee extends boolean = false> = {
   minimumCryptoHuman: string
-  maximumCryptoHuman: string
```
Comment (woodenfurniture) [inline src/lib/swapper/api.ts:156]: Another major focus

File: src/lib/swapper/api.ts [hunk 3]
```diff
@@ -97,7 +97,6 @@ type CommonTradeInput = {
   sellAsset: Asset
   buyAsset: Asset
   sellAmountBeforeFeesCryptoBaseUnit: string
-  sendMax: boolean
```
Comment (woodenfurniture) [inline src/lib/swapper/api.ts:100]: Another major focus

**Comment by woodenfurniture** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4625** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4625" target="_blank">Graphite</a>  👈
    * **PR #4630** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4630" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4625?utm_source=stack-comment).

**Review by 0xApotheosis — APPROVED** [review]
gm

Mainly plumbing changes, though checked quotes for regressions and all is looking tidy.

Onward.

**Pull request merged**